### ISSUE TITLE
Transition doesn't seem to work when using a transition.

### ISSUE THREAD

**Issue by MichMich** [guanzo/vue-smooth-reflow#3]
When using a vue transition to fade to new content, the vue-smooth-reflow effect doesn't seem to work. Do demonstrate the issue, I made a codesandbox example with two example containers, one contains the transition, the other doesn't:

https://codesandbox.io/s/wpmqzk19l

Is there a way around this issue?

**Comment by guanzo** [comment]
Yep, you should use the `transitionEvent` option. It's in the documentation. 

Here's the updated example: https://codesandbox.io/s/k3x4j260qr

Waiting for child transitions to finish is an opt-in feature, meaning it's not done automatically. In the past, it used to be automatic by listening and reacting to any `transitionend` event that fired. In my experience this was a performance killer especially in large component hierarchies, which forced me to rethink how to do it property.

The basic idea of `transitionEvent` is that you register a CSS selector, and/or a CSS property to be matched with a transition event. If it matches, then the smooth reflow will occur. You can see in the example that i set `selector: p`. This is because the elements that are transitioning out are `p` elements. I could have also done this 

```
this.$smoothReflow({
      el: this.$refs.test1,
      transitionEvent: {
          propertyName: 'opacity'
      }
})
```

It would have the same effect, because the elements are transitioning out with `opacity`.

Closing this issue now, hope that was helpful.

**Comment by MichMich** [comment]
Thanks for the extensive explanation.

Unfortunately that doesn't seem to solve the issue. I noticed `transitionEvent` in the documentation and played with it (just like you showed). It didn't gave the desired results. The updated example you mentioned doesn't seem to work either.

Just to be clear: the gray box behind the left content does not reflow smootly, it just jumps from small to large and back (the usual behaviour when vue-smooth-reflow isn't implemented). Are you getting different results?

FYI: I tried both Chrome and Safari.

**Comment by guanzo** [comment]
Hmm... It should definitely work on Chrome. I'm aware that it won't work perfectly on other browsers. My chrome version is:Version 68.0.3440.106 (Official Build) (64-bit). What's yours?

Reopening for now until I can figure it out.

**Comment by MichMich** [comment]
Version 68.0.3440.106 (Official Build) (64-bit) on macOS 10.13.6 (17G65).

**Comment by MichMich** [comment]
Ok, scratch that. I refreshed in Chrome, and now it seems to work. 

It doesn't seem to work in Safari Version 11.1.2 (13605.3.8) though (as you hinted).

**Comment by MichMich** [comment]
FYI: As an experiment, I added the after-leave hook to log to the console, and it seems to work in Safari.

**Comment by KristiyanTs** [comment]
I think I found a workaround.
```
      <component
        :is="loading ? '' : 'transition-group'"
        tag="tbody"
        name="table-row"
      >
        <Task
          v-for="task in tasks"
          :task="task"
          :key="task.id"
        />
      </component>
```
In my example, I use a table but I guess you could use whatever. Then, in the script, after the elements are loaded, set "loading" to false ;)

**Comment by WofWca** [comment]
For me, it looks good in the sandbox you provided (https://codesandbox.io/s/k3x4j260qr), but if I download it and host locally, it does not. The animation is jerky:
1. The old element fades out, the new element gets inserted, starts fading in, height immediately jumps to its final value.
2. The new element finishes fading in, height immediately jumps to its initial value.
3. Height transitions from initial to final.

In my app, however, the behavior is inconsistent – sometimes it behaves as above, sometimes the height just jumps and that's it (like when `vue-smooth-reflow` is not there at all).

`Win 10, Chrome 79.0.3945.88 official build x64`
and
`Win 10, Firefox Developer Edition 72.0b11 (64-bit)`
